File: Kaskus/GridViewPaging/GridViewPagingCs/Pages/DataGenericResult.cs
namespace Dede.App.GridViewPaging.Pages
{
    using System.Collections;
    using System.Collections.Generic;

    public sealed class DataGenericResult<T>
        : DataSourceResult
        where T : class
    {
        public DataGenericResult()
        {
            this.Records = new List<T>();
        }

        public List<T> Records { get; set; }

        protected override IList MyList()
        {
            return this.Records;
        }
    }
}

File: Kaskus/GridViewPaging/GridViewPagingVb/Pages/DataGenericResult.vb
Imports System.Collections
Imports System.Collections.Generic

Namespace Pages

    Public NotInheritable Class DataGenericResult(Of T As Class)
        Inherits DataSourceResult
        Public Sub New()
            Records = New List(Of T)()
        End Sub

        Public Property Records() As List(Of T)

        Protected Overrides Function MyList() As IList
            Return Records
        End Function
    End Class
End Namespace